src/components/JobCard/JobCard.tsx

```diff
@@ -45,15 +45,18 @@ const JobCard = ({ job, idx, pending, pendingJobs, setPendingJobs }: JobCardProp
           setModalVisible(!modalVisible);
         }}>
         <Modal
-          transparent={true}
+          transparent
           visible={modalVisible}
           animationType="slide"
           onRequestClose={() => {
-            setModalVisible(!modalVisible);
+            setModalVisible(false);
```

Comment:
could you try just setting all the instances of !modalVisible in this file to false. i think this should still make the logic work and seems a bit more intuitive.